tests/flagd/pkg/integration/flagd_json_evaluator.go

```diff
@@ -0,0 +1,87 @@
+package integration
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/cucumber/godog"
+	flagd "github.com/open-feature/go-sdk-contrib/providers/flagd/pkg"
+	"github.com/open-feature/go-sdk/pkg/openfeature"
+)
+
+// ctxKeyKey is the key used to pass flag key across context.Context
+type ctxKeyKey struct{}
+
+// ctxKeyKey is the key used to pass the default across context.Context
+type ctxDefaultKey struct{}
+
+// ctxContextKey is the key used to pass the evaluation context across context.Context
+type ctxContextKey struct{}
+
+// ctxContextKey is the key used to pass the value across context.Context
+type ctxValueKey struct{}
+
+func InitializeFlagdJsonScenario(pOptions ...flagd.ProviderOption) func(*godog.ScenarioContext) {
+
+	providerOptions = pOptions
+
+	return initializeFlagdJsonScenario
+}
+
+// initializeFlagdJsonScenario initializes the flagd json evaluator test scenario
+func initializeFlagdJsonScenario(ctx *godog.ScenarioContext) {
+	ctx.Step(`^a flagd provider is set$`, aFlagdProviderIsSet)
+	ctx.Step(`^a string flag with key "([^"]*)" is evaluated with default value "([^"]*)"$`, aFlagdStringFlagWithKeyIsEvaluatedWithDefaultValue)
+	ctx.Step(`^a context containing a key "([^"]*)", with value "([^"]*)"$`, aContextContainingAKeyWithValue)
+	ctx.Step(`^a context containing a nested property with outer key "([^"]*)" and inner key "([^"]*)", with value "([^"]*)"$`, aContextContainingANestedPropertyWithOuterKeyAndInnerKeyWithValue)
+	ctx.Step(`^the returned value should be "([^"]*)"$`, theReturnedValueShouldBe)
+}
```

Comment:
The entire [gherkin test suite](https://github.com/open-feature/flagd-testbed/blob/main/gherkin/flagd-json-evaluator.feature) here is captured by these 4 tests, which are run "matrix style" with the table values.